src/hotspot/share/utilities/growableArray.hpp

```diff
@@ -542,7 +542,7 @@ void GrowableArrayWithAllocator<E, Derived>::shrink_to_fit() {
   this->_capacity = len;        // Must preceed allocate().
   if (len > 0) {
     new_data = static_cast<Derived*>(this)->allocate();
-    for (int i = 0; i < len; ++i) ::new (&new_data[i]) E(old_data[i]);
+    for (int i = 0; i < len; ++i) ::new (&new_data[i]) E(static_cast<E&&>(old_data[i]));
```

Comment:
Why not use `std::move`? Thanks.

Reply:
Hi, in short: Yes, I believe that I can switch and that `std::move` is better.

In long: I can't find any mention of `std::move` not being allowed in the style guide, using it seems to be in the same spirit as other std library usage (`#include <new>` for example) and since it is just a better version of what I wrote here let's do the switch.

Reply:
I believe "Rvalue references and move semantics" falls under the "Undecided Features" category.

Reply:
@jdksjolen  The guide lists what is permitted to be used, so if it isn't listed as permitted then it isn't presently permitted.

Reply:
Thank you David, then I guess I'd need to get the guide changed before my or Axel's proposal is merged.

Reply:
@merykitty is correct, at the bottom of the style guide under undecided features:

> Rvalue references and move semantics